UserSettingsPage: device fingerprint on its own line

adasauce opened the pull request:
pinephone users have an unusable settings page, as the device fingerprint stops the layout from shrinking enough to fit the form toggles into view.

the `deviceFingerprintValue_` is now on a new line, and the font size for `monospaceFont` has been slightly reduced. this also makes the device ID label a smidge smaller, but IMO it looks like it fits.

now the thing hitching the layout is the dropdowns for font selection. it seems like this could probably be solved with a more cohesive mobile / responsive design than fixing individual widgets. but I think maybe its worth making the settings screen usable on mobile at all vs. waiting for a big refactor.

I realize all this discussion happened in the matrix room without any maintainers around, happy to hear any feedback or whether you have alternative plans to address this.

as of 0.7.0-dev look:
![image](https://user-images.githubusercontent.com/60991921/74398825-3500f880-4def-11ea-98f2-832288e575b8.png)

new look:
![image](https://user-images.githubusercontent.com/60991921/74398746-06831d80-4def-11ea-9055-b8b3c517f8a5.png)

deepbluev7 commented:
So, I don't think this is the right way to go. This changes the fingerprint to always be in the next line. I don't think we want that on the desktop, especially since it doesn't fit with the rest of the style of the settings page. I think we should really implement a proper responsive design for the settings window instead. The easy way would be to make each setting individually overflow, the better way would be to let every setting overflow, if a single settings doesn't fit in one line. One way to do that, would be to implement a layout like this: https://doc.qt.io/qt-5/qtwidgets-layouts-flowlayout-example.html

That may be too complicated for you, if you don't have much experience with Qt. We also want to port Nheko entirely to Qml at some point, which would make this work currently redundant. Qml gives you a lot more tools to make a layout responsive though, so it would probably fix most issues you are having. Some additional questions I have, since I don't own a pinephone:

- Can you scroll the settings by dragging the layout or do you have to use the scrollbar? (Qml allows that by default)
- How do other applications handle drop downs? (Maybe the Qml dropdowns work for that)

Alternatively, you could try (as a quick fix) to reduce the margin of the settings widget from 0.2 to 0.1 or 0.05. Since you are not missing that much space in the settings widget, that may be enough for you to make the window usable?

adasauce commented:
> So, I don't think this is the right way to go. This changes the fingerprint to always be in the next line. I don't think we want that on the desktop, especially since it doesn't fit with the rest of the style of the settings page. I think we should really implement a proper responsive design for the settings window instead. The easy way would be to make each setting individually overflow, the better way would be to let every setting overflow, if a single settings doesn't fit in one line. One way to do that, would be to implement a layout like this: https://doc.qt.io/qt-5/qtwidgets-layouts-flowlayout-example.html

Agreed that responsive is the best way. The flow layout looks pretty straightforward, I could give it a try -- what's the worst that could happen :laughing: 

> That may be too complicated for you, if you don't have much experience with Qt. We also want to port Nheko entirely to Qml at some point, which would make this work currently redundant. Qml gives you a lot more tools to make a layout responsive though, so it would probably fix most issues you are having. Some additional questions I have, since I don't own a pinephone:
> 
>     Can you scroll the settings by dragging the layout or do you have to use the scrollbar? (Qml allows that by default)
>     How do other applications handle drop downs? (Maybe the Qml dropdowns work for that)

unfortunately, I don't have my pinephone yet :) it didn't get shipped out before chinese new year, so I'll probably still be waiting another month or so. 

I implemented this change on behalf of `grossaffe` in the matrix room who was having a bunch of UX issues on his phone. this settings issue was his first blocker to using the app. if he's still around he could probably answer the phone specific questions. I was able to recreate his settings layout issue by just making the window to that of his device size, and implemented based on that.

> Alternatively, you could try (as a quick fix) to reduce the margin of the settings widget from 0.2 to 0.1 or 0.05. Since you are not missing that much space in the settings widget, that may be enough for you to make the window usable?

I can give this margin change a shot to see how it looks. I think it might make the settings screen on the desktop look very wide though.

deepbluev7 commented:
If it looks too wide with a smaller margin, it would probably look too wide on a 4k screen too... The layouting logic of the settings page really needs some work!

Christian-Gibbons commented:
The primary issue the pinephone has--at least on PostmarketOS with Plasma Mobile--is that the display size is improperly detected.  Looking at the nheko.conf file after running nheko, it shows the window height as 666 pixels, and the width as 790 pixels.  The phone screen's true dimensions are 1440 pixels tall by 720 pixels wide.

So what's happening in practice is that the window is being drawn beyond the edges of the screen by a significant amount.  All of the configuration settings are unreachable, which is an annoyance, but at least those can probably be set manually in the config file.  The biggest problems are that the login screen's login-button cannot be reached (I managed to get around this by copying my credentials over from my desktop), and that the main chat window is drawn almost entirely off-screen with no way to scroll.  I will attempt to post a picture of it when I get a chance.

adasauce commented:
Handling in #122 using a different approach.
